forward/ft_apply_montage.m

```diff
@@ -95,6 +95,13 @@
 haschantype = (isfield(input, 'chantype') || isfield(input, 'chantypenew')) && all(isfield(montage, {'chantypeold', 'chantypenew'}));
 haschanunit = (isfield(input, 'chanunit') || isfield(input, 'chanunitnew')) && all(isfield(montage, {'chanunitold', 'chanunitnew'}));
 
+if ~istrue(keepunused)
+  % remove the channels that are not rereferenced from the input
+  cfg = [];
+  cfg.channel = montage.labelold;
+  input = ft_selectdata(cfg, input);
```

Comment:
this will fail on many of the input structures that ft_apply_montage accepts. Not all of those input structures are correctly dealt with in ft_selectdata.